docs/how-to-guides/admin/gitlab.rst

```diff
@@ -0,0 +1,240 @@
+#. Create a file with the connection details for the various non-backup Gitlab S3 buckets using the `registry.s3.yaml example provided by GitLab <https://gitlab.com/gitlab-org/charts/gitlab/-/blob/master/examples/objectstorage/registry.s3.yaml>`_. If connection details vary between buckets, multiple files can be created for each connection. `GitLab's bucket connection documentation <https://docs.gitlab.com/charts/charts/globals.html#connection>`_
+
+Migration
+^^^^^^^^^
+
+#. Uninstall Renku & modify your Renku Helm values to use an external GitLab, specifying the new GitLab URL. You can also start with a newly generated Renku values using the `Renku generate-values script <https://github.com/SwissDataScienceCenter/renku/tree/master/scripts/generate-values>`_.
+#. (Optional, recommended) Create a new namespace for GitLab
+#. Create secrets in the namespace you wish to deploy GitLab to with the content of the files containing the S3 bucket connection details and OAuth2 provider. Ensure the name of the secret and the key of the values match the associated values provided in your cloud-native Helm chart values.
+#. Install GitLab with Helm, providing the namespace, chart version and values file as arguments.
+#. After GitLab has successfully installed, restore the backup of your existing GitLab instance by following `Gitlab's restoring a GitLab installation documentation <https://docs.gitlab.com/charts/backup-restore/restore.html>`_, starting with restoring the rails secrets. There might be errors or warnings during the restore, but according to `this GitLab issue <https://gitlab.com/gitlab-org/gitlab/-/issues/266988>`_, that is often to be expected, and does not mean that the restore has actually failed.
+#. Ensure you can log in to GitLab, even just as root user, and do some checks to ensure the restore was successful.
+
+Upgrading Renku with the newly modified Helm values
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Backup your current unedited values file
+#. Replace every GitLab URL from \https://$RENKU_URL/gitlab to \https://gitlab.$RENKU_URL. There should be 4 instances, at ``gateway.gitlabUrl``, ``graph.gitlab.url``, ``notebooks.gitlab.url`` and ``ui.gitlabUrl``.
+#. If you have a value set at ``global.gitlab.urlPrefix`` change it from ``/gitlab`` to ``/``
+#. Set ``gitlab.enabled`` to ``false``.
+#. Re-install the Renku Helm chart with the newly modified values.
+
+Troubleshooting
+^^^^^^^^^^^^^^^
+
+It is possible there will be some issues post-upgrade.
+
+If you can sign in normally to GitLab using a Keycloak account, but Renku still shows the ``RenkuLab is down`` message, try deploying a Renku instance with a newly generated values file to a different namespace and see if the issue persists. If the new Renku instance is able to launch correctly, there is probably an issue with your Renku configuration, in particular the connection to GitLab.
```

Comment:
I would remove this whole `Troubleshooting` section. It is short and it may not be feasible for people to deploy Renku again in a different namespace.